code/datums/components/autofire.dm

```diff
@@ -59,6 +60,11 @@
 	SIGNAL_HANDLER
 	auto_fire_shot_delay = _auto_fire_shot_delay
 
+///Setter for autoburst shot delay
+/datum/component/automatedfire/autofire/proc/modify_autoburst_shot_delay(datum/source, _auto_burst_fire_shot_delay)
+	SIGNAL_HANDLER
+	auto_burst_fire_shot_delay = _auto_burst_fire_shot_delay
```

Comment:
if you don't allow for a direct set in the constructor, it seems weird to allow it here